### PR TITLE
[DOC] link surface terms to glossary

### PR DESCRIPTION
**Pull request by Remi-Gau**
- closes https://github.com/nilearn/nilearn/issues/4152

Changes proposed in this pull request:

- add surface related terms to glossary (vertex, face, mesh) and links them in doc and codebase

### CODE REVIEW

File: doc/glossary.rst [hunk 1]
```diff
@@ -90,6 +90,10 @@ If you wish to add a missing term, please
         Erosion uses a structuring element for probing and reducing the shapes
         contained in the input image.
 
+    faces
+        When referring to surface data, a face corresponds to one of the triangle
```
Comment (ymzayek) [inline doc/glossary.rst:94]:
```suggestion
        When referring to surface data, a face corresponds to one of the triangles
```

File: nilearn/experimental/surface/_surface_image.py [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
 
 class Mesh(abc.ABC):
-    """A surface mesh, having vertex coordinates and faces(triangles)."""
+    """A surface mesh, having vertex coordinates and faces (triangles)."""
```
Comment (ymzayek) [inline nilearn/experimental/surface/_surface_image.py:17]:
```suggestion
    """A surface mesh having vertex, coordinates and faces (triangles)."""
```

Comment (ymzayek) [reply nilearn/experimental/surface/_surface_image.py:17]: why not link the terms here too?

File: doc/sg_execution_times.rst
Comment (ymzayek) [inline doc/sg_execution_times.rst:1]: Why add this file? This is usually produced for each example anyways when built and can be viewed with this pattern e.g. `https://nilearn.github.io/stable/auto_examples/08_experimental/sg_execution_times.html` for each example although it's essentially "hidden" on the website. I don't think we need to add it right?

**Review by ymzayek — COMMENTED** [review]
Thank @Remi-Gau !
I wonder if we should add 'texture'? It's not used a lot in the docs but more so as a variable name when it equals an array projected on the surface

File: doc/sg_execution_times.rst
Comment (Remi-Gau) [reply doc/sg_execution_times.rst:1]: yup that's a "misfire" from running an example. Sorry about that.

File: nilearn/experimental/surface/_surface_image.py (hunk 2 above)
Comment (Remi-Gau) [reply nilearn/experimental/surface/_surface_image.py:17]: I missed it. Was working late last night. 🙈

**Comment by Remi-Gau** [comment]
> Thank @Remi-Gau ! I wonder if we should add 'texture'? It's not used a lot in the docs but more so as a variable name when it equals an array projected on the surface

yes definitely

I had listed also:
- coordinates
- geometry

But they feel too generic, no?

**Comment by Remi-Gau** [comment]
> if we should add 'texture'?

Only mentioned once: nilearn/examples/01_plotting/plot_3d_map_to_surface_projection.py:53

So maybe not?

**Comment by bthirion** [comment]
No strong opinion on "textures". We could also use it more frequently...

**Comment by Remi-Gau** [comment]
Would suggest to merge this as is and add texture later

**Comment by bthirion** [comment]
Good with me.

**Pull request merged**